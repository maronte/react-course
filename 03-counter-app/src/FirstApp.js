import React, { Fragment } from "react";
import { PropTypes } from "prop-types";
// My first functional component.

// Props always are passed in as an object.
const FirstApp = ({ greet, name }) => {

    return (
        <Fragment>
            <h1>{ greet } { name }</h1>
            { /** You can add expressions or variables that are strings or numbers, also if you add
             * an array, it will be rendered as a list with /n as glue, the array also contains same data types.
             * Other objects are not supported. 
             * */ }
            <p>This is my first React app.</p>
        </Fragment>
    );

    // Also you can use a Fragment writting <> </> and it will be a Fragment without implicit import.

}


// PropTypes are used to validate the props.
FirstApp.propTypes = {
    name: PropTypes.string.isRequired
};

// Also you can add default props. Different from destructuring props, it is property is added the object.

FirstApp.defaultProps = {
    greet: "Hello"
}

export default FirstApp;