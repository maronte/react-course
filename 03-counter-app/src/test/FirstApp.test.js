import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import FirstApp from "../FirstApp";
import "../setupTests";

describe("FirstApp component", () => {
    /*test("should shows a message with greeting", () => {
        const greeting = "Hello John";

        const view = render(<FirstApp name="John" />);

        expect(view.getByText(greeting)).toBeInTheDocument();
    });*/

    test("should show the component", () => {
        const wrapper = shallow(<FirstApp name="John" />);
        expect(wrapper).toMatchSnapshot();
    });

    test("should show the component with a greeting", () => {
        const expectedGreeting = "Hello John";
        const wrapper = shallow(<FirstApp name="John" />);
        const greeting = wrapper.find("h1").text();
        expect(greeting).toBe(expectedGreeting);
    });
});