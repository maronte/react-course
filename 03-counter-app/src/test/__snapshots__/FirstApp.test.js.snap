// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp component should show the component 1`] = `
<Fragment>
  <h1>
    Hello
     
    John
  </h1>
  <p>
    This is my first React app.
  </p>
</Fragment>
`;
